diffsync/utils.py

```diff
@@ -42,3 +47,44 @@ def __missing__(self, key):
         """When trying to access a nonexistent key, initialize the key value based on the internal factory."""
         self[key] = value = self.factory()
         return value
+
+
+# from: https://stackoverflow.com/questions/72618673/list-directory-tree-structure-in-python-from-a-list-of-path-file
+def _tree(data: dict, prefix: str = ""):
+    """Given a dictionary will yield a visual tree structure.
+
+    A recursive generator, given a dictionary will yield a visual tree structure line by line
+    with each line prefixed by the same characters.
+    """
+    # contents each get pointers that are ├── with a final └── :
+    pointers = [TEE] * (len(data) - 1) + [LAST]
+    for pointer, path in zip(pointers, data):
+        yield prefix + pointer + path
+        if isinstance(data[path], dict):  # extend the prefix and recurse:
+            extension = BRANCH if pointer == TEE else SPACE
+            # i.e. SPACE because LAST, └── , above so no more |
+            yield from _tree(data[path], prefix=prefix + extension)
+
+
+def tree_string(data: dict, root):
```

Comment:
We need to use typing.Dict rather than builtins.dict as long as we have to support a Python version <3.9